pkg/meta/meta.go

```diff
@@ -62,110 +34,63 @@ func Mirror() string {
 	return repository.DefaultMirror
 }
 
-// InitRepository initialize the repository
-func InitRepository(options repository.Options) error {
+// Environment is the user's fundamental configuration including local and remote parts.
+type Environment struct {
+	// profile represents the TiUP local profile
+	profile *localdata.Profile
+	// repo represents the components repository of TiUP, it can be a
+	// local file system or a HTTP URL
+	repo *repository.Repository
+}
+
+// InitEnv creates a new Environment object configured using env vars and defaults.
+func InitEnv(options repository.Options) (*Environment, error) {
+	profile := localdata.InitProfile()
+
 	// Initialize the repository
 	// Replace the mirror if some sub-commands use different mirror address
 	mirror := repository.NewMirror(Mirror(), repository.MirrorOptions{})
-	r, err := repository.NewRepository(mirror, options)
-	repo = r
-	return err
+	repo, err := repository.NewRepository(mirror, options)
+	return &Environment{profile, repo}, err
+}
+
+// New creates a new Environment with the provided data.
+func New(profile *localdata.Profile, repo *repository.Repository) *Environment {
+	return &Environment{profile, repo}
 }
 
 // Repository returns the initialized repository
-func Repository() *repository.Repository {
-	return repo
+func (env Environment) Repository() *repository.Repository {
+	return env.repo
 }
 
 // SetRepository exports for test
-func SetRepository(r *repository.Repository) {
-	repo = r
+func (env Environment) SetRepository(r *repository.Repository) {
+	env.repo = r
 }
 
 // Profile returns the profile of local data
-func Profile() *localdata.Profile {
-	return profile
+func (env Environment) Profile() *localdata.Profile {
+	return env.profile
 }
 
 // SetProfile exports for test
-func SetProfile(p *localdata.Profile) {
-	profile = p
+func (env Environment) SetProfile(p *localdata.Profile) {
```

Comment:
The receiver must a pointer, or else the `set` will take no effect.
```suggestion
func (env *Environment) SetProfile(p *localdata.Profile) {
```